### PR TITLE
[WIP] Publish to preview stream 

### PR DESCRIPTION
**Pull request by Reettaphant**
- Publish media atom to preview kinesis stream when new content is added to it

@paulmr

### CODE REVIEW

File: app/controllers/Api.scala [hunk 1]
```diff
@@ -69,7 +71,16 @@ class Api @Inject() (val dataStore: DataStore,
...
+          _ => {
+            Created(s"updated atom $atomId").withHeaders("Location" -> atomUrl(atom.id))
+
+            val event = ContentAtomEvent(atom, EventType.Update, now())
+
+            previewPublisher.publishAtomEvent(event) match {
+              case Success(_)  => NoContent
+              case Failure(err) => InternalServerError(jsonError(s"could not publish: ${err.toString}"))
+            }
+          }
```
Comment (jamesgorrie) [inline app/controllers/Api.scala]: Should we be doing this in `createMediaAtom` too?

Comment (Reettaphant) [inline app/controllers/Api.scala]: Yes I think so. I forgot all about that.

**Comment by paulmr** [comment]
Let's add the new liveStream and previewStream settings to the `reference.conf` file as a pointer to users that they need to be filled in.

Also we will need to update the configuration files on S3 that are used for deployment. (Just stating that as a reminder for after this is deployed, not something we need to do as part of the PR).

File: app/controllers/Api.scala [hunk 2]
```diff
@@ -69,7 +80,16 @@ class Api @Inject() (val dataStore: DataStore,
 
         dataStore.updateMediaAtom(newAtom).fold(
           err => InternalServerError(err.msg),
-          _ => Created(s"updated atom $atomId").withHeaders("Location" -> atomUrl(atom.id))
+          _ => {
+            
+            val event = ContentAtomEvent(atom, EventType.Update, now())
```
Comment (paulmr) [inline app/controllers/Api.scala]: I think this should be `newAtom`? Current it is publishing the state of the atom before the asset is added.

File: test/ApiSpec.scala [hunk 3]
```diff
@@ -121,11 +128,21 @@ class ApiSpec
...
       }
     }
-    "call out to publisher to publish an atom" in withApi() { api =>
+    "call out to live publisher to publish an atom" in withApi() { api =>
       val result = call(api.publishAtom("1"), requestWithCookies(api))
       status(result) mustEqual NO_CONTENT
     }
-    "call report failure if publisher fails" in withApi(publisher = failingMockPublisher) { api =>
+
+    "call out to preview publisher when adding an asset" in withApi() { api =>
```
Comment (paulmr) [inline test/ApiSpec.scala]: Let's add a test here to catch the situation mentioned in the above comment: so we can use the mock publisher `verify()` to confirm that the atom update includes the new asset.

**Comment by paulmr** [comment]
Nice work! :+1:

**Pull request merged**